pkg/controller/endpointslice/reconciler.go

```diff
@@ -173,7 +173,19 @@ func (r *reconciler) reconcileByAddressType(service *corev1.Service, pods []*cor
 
 		node, err := r.nodeLister.Get(pod.Spec.NodeName)
 		if err != nil {
-			return err
+			// The Node can not be present in the informer cache, but ShouldPodBeInEndpointSlice guarantee that the Pod has IPs,
+			// so we know that the Node has already existed. We don't know if the Node has gone permanently or temporary.
+			// If this is an informer error or yhr topologyCache is not available we exit immediately, keeping the previous fail-open behavior,
+			// the slice is not updated and it keeps  requeueing this Service until the Pod is definitively deleted by the PodGC controller, the Node recovers,
+			// or until the maxBackoff retry timeout.
+			// If the topology cache is enabled, the Node update/create/delete events will be processed and this Service will be reconciled,
+			// so we can skip this Pod and fail-close not sending traffic to the Pod that at this moment we don't know its real status.
```

Comment:
I don't think this is accurate. The resync will only apply to Services that have hints enabled _and_ set. There's actually likely a bug here that adding a node does not seem like it would result in hints being set even if that brought us to an ideal ratio. Even with that fixed, this would not solve the problem for all Services without hints enabled.

https://github.com/kubernetes/kubernetes/blob/a1ac74224eac04dae21fcee2b32f8e914111ec9f/pkg/controller/endpointslice/endpointslice_controller.go#L544-L548

That relies on a Service being present in `t.endpointsByService`:
https://github.com/kubernetes/kubernetes/blob/a1ac74224eac04dae21fcee2b32f8e914111ec9f/pkg/controller/endpointslice/topologycache/topologycache.go#L73

Anytime we can't get a reasonable allocation for each zone, we remove hints:
https://github.com/kubernetes/kubernetes/blob/a1ac74224eac04dae21fcee2b32f8e914111ec9f/pkg/controller/endpointslice/topologycache/topologycache.go#L138-L143

Removing hints removes the Service from `t.endpointsByService`:
https://github.com/kubernetes/kubernetes/blob/a1ac74224eac04dae21fcee2b32f8e914111ec9f/pkg/controller/endpointslice/topologycache/topologycache.go#L166

Reply:
ok, then my assumption was wrong, I was thinking it will queue any service related to that node